realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -346,14 +312,56 @@ protected void onError(LogoutResponse response) {
     }
 
     /**
+     * Change this users password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
```

Comment:
I didn't test it this time around, but Android has crashed for some years now if you do networking on the UI thread. I'll add a unit test to verify.